src/app/views/NoteCreate.svelte

```diff
@@ -101,18 +105,24 @@
         (options.publish_at && Math.floor(options.publish_at.getTime() / 1000)) || undefined,
     })
 
-    const event = await sign(template, options)
-
     drafts.set("notecreate", $editor.getHTML())
 
+    publishing = "signing"
+
+    let event = await sign(template, options)
```

Comment:
Signing has to happen after pow generation. Did you test this?

Reply:
Yeah with the nonce in the tags you probably need to sign it again
The sign function attach the pubkey to the event, but the signature is likely invalid once the nonce is added

Reply:
> Signing has to happen after pow generation. Did you test this?

note1lxkagxz9mpg8m85j78qepfq3l44dpt03qtqrm23nxc78yayv9chs7cs6dn

This was my latest test

Reply:
You can use `stamp`, `hash` and `own` from @welshman/signer for this, a redundant signature is pretty wasteful especially over nip 46. Also, that note doesn't have a pow tag.

Reply:
the PoW tag is added by the pow worker. 
I used to attach the nonce as a field on the event (which is wrong), that's why it was fine to sign before running the PoW, now that the nonce is in the tag, as it should, the serialisation change. 
I will sign once after the pow is done (eventually)